Source/modules/webcl/WebCLMemoryUtil.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2014 Samsung Electronics Corporation. All rights reserved.
+// Copyright (C) 2015 Intel Corporation All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "config.h"
+
+#if ENABLE(WEBCL)
+
+#include "modules/webcl/WebCL.h"
+#include "modules/webcl/WebCLBuffer.h"
+#include "modules/webcl/WebCLCommandQueue.h"
+#include "modules/webcl/WebCLContext.h"
+#include "modules/webcl/WebCLDevice.h"
+#include "modules/webcl/WebCLKernel.h"
+#include "modules/webcl/WebCLMemoryObject.h"
+#include "modules/webcl/WebCLMemoryUtil.h"
+#include "modules/webcl/WebCLObject.h"
+#include "modules/webcl/WebCLOpenCL.h"
+#include "modules/webcl/WebCLProgram.h"
+
+namespace blink {
+
+const char* programSource = \
+    "__kernel void init(__global char* buffer, unsigned offset, unsigned count) "\
+    "{                                                                          "\
+    "    unsigned i = get_global_id(0);                                         "\
+    "    unsigned displacedI = i + offset;                                      "\
+    "    if (displacedI < count)                                                "\
+    "        buffer[displacedI] = (char)(0);                                    "\
+    "}                                                                          "\
+    "__kernel void init16(__global char16* buffer, unsigned count)              "\
+    "{                                                                          "\
+    "    unsigned i = get_global_id(0);                                         "\
+    "    if (i < count)                                                         "\
+    "        buffer[i] = (char16)(0);                                           "\
+    "}                                                                          ";
+
+WebCLMemoryUtil::WebCLMemoryUtil(WebCLContext* context)
+    : m_context(context)
+{
+}
+
+WebCLMemoryUtil::~WebCLMemoryUtil()
+{
+}
+
+void WebCLMemoryUtil::ensureMemory(WebCLMemoryObject* memoryObject, WebCLCommandQueue* commandQueue, ExceptionState& es)
+{
+    cl_int err = CL_SUCCESS;
+    // The program and kernels are used to intialize OpenCL memory to 0.
+    // Every created OpenCL memory should call this function to intialize.
```

Comment:
No detail data on this. But from the real case, it don't show performance decrease.
WebGL solution is better. After WebCL validator is integrated, this part code could be removed.

I would add comments for it. Thanks